frontend/src/metabase/sharing/components/EmailAttachmentPicker.unit.spec.js

```diff
@@ -0,0 +1,198 @@
+import React from "react";
+import { render, fireEvent, screen } from "@testing-library/react";
+
+import EmailAttachmentPicker from "./EmailAttachmentPicker";
+
+describe("EmailAttachmentPicker", () => {
+  describe("when instantiated without any cards with attachments", () => {
+    let pulse;
+    let setPulse;
+    beforeEach(() => {
+      pulse = createPulse();
+      setPulse = jest.fn();
+      render(
+        <EmailAttachmentPicker
+          cards={pulse.cards}
+          pulse={pulse}
+          setPulse={setPulse}
+        />,
+      );
+    });
+
+    it("should have a Toggle that is not toggled", () => {
+      const toggle = screen.getByRole("checkbox");
+      expect(toggle).toBeInTheDocument();
+      expect(toggle.dataset.testid).toEqual("toggle");
+      expect(toggle.getAttribute("aria-checked")).toBe("false");
+    });
+
+    it("should have a clickable toggle that reveals attachment type and a checkbox per question", () => {
+      expect(screen.queryByText("File format")).toBeNull();
+      expect(screen.queryByText("Questions to attach")).toBeNull();
+      expect(screen.queryByText("card1")).toBeNull();
+      expect(screen.queryByText("card2")).toBeNull();
+
+      const toggle = screen.getByRole("checkbox");
+      fireEvent.click(toggle);
+
+      const csvFormatInput = screen.getByLabelText(".csv");
+      expect(csvFormatInput.checked).toBe(true);
+
+      const toggleAllCheckbox = screen.getByLabelText("Questions to attach");
+      expect(toggleAllCheckbox.checked).toBe(false);
+
+      const card1Checkbox = screen.getByLabelText("card1");
+      expect(card1Checkbox.checked).toBe(false);
+
+      const card2Checkbox = screen.getByLabelText("card2");
+      expect(card2Checkbox.checked).toBe(false);
```

Comment:
Guess we can use `expect(checkbox).toBeChecked()` and `expect(checkbox).not.toBeChecked()` here